Automattic/sensei
# Remove pagination from legacy course results page

Fixes #3965

### Changes proposed in this Pull Request

The issue is that in some themes the legacy "Course Results" page renders unwanted pagination at the bottom. The issue is not present with the latest version of Sensei plugin. However, updating the plugin does not fix the issue, because the plugin keeps using the legacy "Course Results" page. It is possible for user to fix this issue manually by creating a new page, rendering "Course Results" block in it and assigning that page to "Course Completed Page" field in Sensei LMS Settings. However the older versions of Sensei will have this error by default.

This change simply does not render pagination for the legacy "Course Results" page.

### Testing instructions


To reproduce the bug:
- Setup wordpress site.
- Install and activate StoreFront theme.
- Create ~30 dummy posts. You can use [Lorem Ipsum](https://wordpress.org/plugins/wp-lorem-ipsum/) plugin for this.
- Install Sensei version <= [3.12.0](https://github.com/Automattic/sensei/releases/tag/version%2F3.12.0).
- Activate the plugin and install sample course during the setup wizard.
- Add "View Results" block into the sample course page.
- Visit the sample course page and take the course.
- Click on the View Results button.
- Confirm that there is unwanted pagination at the bottom. ⚠️ 
- Go to plugins page and update Sensei to the latest version.
- Go to the the sample course page and click "View Restuls" again.
- Confirm that there is still an unwanted pagination at the bottom of the "Course Results" page.

To confirm the fix:
- Replace the Sensei plugin directory with the one from [this branch](https://github.com/Automattic/sensei/archive/refs/heads/fix/course-results-pagination.zip).
- Refresh the "Course Results" page.
- Confirm the unwanted pagination is done. ✅ 


### New/Updated Hooks

- `sensei_pagination` no longer renders pagination for legacy "Course Results" page.


### Deprecated Code

- n/a


### Screenshot / Video

- n/a
